src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax.model;
+
+import java.util.Arrays;
+
+public class FruitTransaction {
+    public enum Operation {
+        BALANCE("b"), SUPPLY("s"), PURCHASE("p"), RETURN("r");
+
+        private final String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public static Operation fromCode(String code) {
```

Comment:
The method `fromCode` is a good use of the enum pattern, but consider moving this logic to a separate service or utility class if it's going to be used in multiple places. This will adhere to the Single Responsibility Principle (part of SOLID).